src/sentry/utils/sentry_apps/webhooks.py

```diff
@@ -126,69 +127,87 @@ def send_and_save_webhook_request(
     :param url: The URL to hit for this webhook if it is different from `sentry_app.webhook_url`.
     :return: Webhook response
     """
-    buffer = SentryAppWebhookRequestsBuffer(sentry_app)
+    from sentry.sentry_apps.metrics import SentryAppInteractionEvent, SentryAppInteractionType
 
-    org_id = app_platform_event.install.organization_id
     event = f"{app_platform_event.resource}.{app_platform_event.action}"
-    slug = sentry_app.slug_for_metrics
-    url = url or sentry_app.webhook_url
-    assert url is not None
-
-    try:
-        response = safe_urlopen(
-            url=url,
-            data=app_platform_event.body,
-            headers=app_platform_event.headers,
-            timeout=options.get("sentry-apps.webhook.timeout.sec"),
-        )
-    except (Timeout, ConnectionError) as e:
-        error_type = e.__class__.__name__.lower()
-        logger.info(
-            "send_and_save_webhook_request.timeout",
-            extra={
-                "error_type": error_type,
-                "organization_id": org_id,
-                "integration_slug": sentry_app.slug,
-                "url": url,
-            },
-        )
-        track_response_code(error_type, slug, event)
+    with SentryAppInteractionEvent(
+        operation_type=SentryAppInteractionType.SEND_WEBHOOK, event_type=SentryAppEventType(event)
```

Comment:
should we try to cast `event` to `SentryAppEventType` outside of the context manager and catch errors if we are unable to?